### ISSUE TITLE
Figure directive wrapped in confluence information macro

### ISSUE THREAD

**Issue by Bartcardi** [sphinx-contrib/confluencebuilder#598]
Hello! 

When I use a figure directive to insert an image to my confluence page, it gets wrapped in an information macro. Is there a way to disable this? When I use specific figure dimensions the information macro does not adjust to those dimensions. Below is an example. I have also inserted the `sphinx-build-confluence report output`

<img width="1307" alt="Screenshot 2021-12-07 at 15 49 00" src="https://user-images.githubusercontent.com/11993543/145051056-d49b26be-ead7-4431-99ee-a46d267935a8.png">

```
(system)
 platform: Linux-5.10.47-linuxkit-x86_64-with-glibc2.17
   python: 3.8.12 (default, Oct 12 2021, 13:49:34) [GCC 7.5.0]
   sphinx: 4.3.0
 requests: 2.26.0
  builder: 1.7.1

(configuration)
confluence_page_hierarchy: True
confluence_parent_page: (set)
confluence_publish: True
confluence_server_pass: (set)
confluence_server_url: https://<removed> (cloud)
confluence_server_user: (set)
confluence_space_key: (set; upper)

(confluence instance)
 connected: yes
   fetched: yes
   decoded: yes
    parsed: yes
      type: confluence
   version: 1000.0.0-c98c4c609e2b
     build: 6452
```

**Comment by jdknight** [comment]
@Bartcardi, at this time, figures are always wrapped with the macro. I cannot seem to recall why the choice was made, but I believe it was either a styling preference at the time and/or something to deal with floating/alignment of images. Since this styling is not really consistent with most themes and there are alternative ways to deal with floating, this styling should be removed.

In the interim, a (quickly made) workaround to remove this macro usage, the following can be added into a documentation's configuration:

```
from docutils import nodes

def custom_visit_figure(self, node):
    self.body.append('<p>')

def custom_depart_figure(self, node):
    self.body.append('</p>')
    self.body.append('<div style="clear: both"> </div>\n')

def setup(app):
    app.add_node(nodes.figure, confluence=(custom_visit_figure, custom_depart_figure), override=True)
```

**Comment by Bartcardi** [comment]
Thank you very much for your quick response. That's a very nifty trick and works beautifully. For me this closes the issue. I should probably monitor future release notes that address this issue. 

I was wondering if I could extend on this workaround to have the sphinx admonitions look more like the confluence Info / Note / Success / Warning / Error panels. Do you think this is possible?

**Comment by jdknight** [comment]
> I was wondering if I could extend on this workaround to have the sphinx admonitions look more like the confluence Info / Note / Success / Warning / Error panels. Do you think this is possible?

If this is in reference to the admonitions on Confluence new editor, unfortunately, it is not possible at this time. Both storage format for v1 display (what this extension currently produces) and ADF format, each define (for example) an "info" macro -- and each style is rendered in their own ways. There is work being done to support publishing in an ADF format (#347), but it is a work-in-progress (maybe early next year). There is a lot effort to support as much of the existing capability on the new format (without getting into limitations of the new format).

**Comment by Bartcardi** [comment]
I realised it was a bit of a naive question after going through some of the issues, thanks for the response though, and thank you for the great work so far. We are really happy using this in our project's workflow.

**Comment by jdknight** [comment]
Changes have been added in #605 to remove the undesired macro usage on figure types. This change has been added into the main branch and should be made available next stable release (v1.8). Users affected by this issue and desire a fix before a release is made, the [development version](https://sphinxcontrib-confluencebuilder.readthedocs.io/en/stable/install/#development-installation) of this extension can be installed or the [workaround mentioned above](https://github.com/sphinx-contrib/confluencebuilder/issues/598#issuecomment-988381979) can be used.

**Comment by jdknight** [comment]
v1.8 is now available on PyPI -- marking as closed.
